contracts/factory/RegistryWhitelistFactory.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+// ──────────────────────────────────────────────────────────────────────────────
+//     _   __    _  __
+//    / | / /__ | |/ /_  _______
+//   /  |/ / _ \|   / / / / ___/
+//  / /|  /  __/   / /_/ (__  )
+// /_/ |_/\___/_/|_\__,_/____/
+//
+// ──────────────────────────────────────────────────────────────────────────────
+// Nexus: A suite of contracts for Modular Smart Accounts compliant with ERC-7579 and ERC-4337, developed by Biconomy.
+// Learn more at https://biconomy.io. To report security issues, please contact us at: security@biconomy.io
+
+import { LibClone } from "solady/src/utils/LibClone.sol";
+import { BytesLib } from "../lib/BytesLib.sol";
+import { INexus } from "../interfaces/INexus.sol";
+import { BootstrapConfig } from "../utils/RegistryBootstrap.sol";
+import { AbstractNexusFactory } from "./AbstractNexusFactory.sol";
+import { IERC7484 } from "../interfaces/IERC7484.sol";
+import { MODULE_TYPE_VALIDATOR, MODULE_TYPE_EXECUTOR, MODULE_TYPE_FALLBACK, MODULE_TYPE_HOOK } from "../types/Constants.sol";
+
+/// @title RegistryFactory
+/// @notice Factory for creating Nexus accounts with whitelisted modules. Ensures compliance with ERC-7579 and ERC-4337 standards.
+/// @author @livingrockrises | Biconomy | chirag@biconomy.io
+/// @author @aboudjem | Biconomy | adam.boudjemaa@biconomy.io
+/// @author @filmakarov | Biconomy | filipp.makarov@biconomy.io
+/// @author @zeroknots | Rhinestone.wtf | zeroknots.eth
+/// Special thanks to the Solady team for foundational contributions: https://github.com/Vectorized/solady
+contract RegistryFactory is AbstractNexusFactory {
+    /// @notice Error thrown when a non-whitelisted module is used.
+    /// @param module The module address that is not whitelisted.
+    error ModuleNotWhitelisted(address module);
+
+    /// @notice Error thrown when a zero address is provided.
+    error ZeroAddressNotAllowed();
+
+    IERC7484 public immutable REGISTRY;
+    address[] public attesters;
```

Comment:
this would maintain attesters here instead of calling trust attesters from the account?

Reply:
in this case, the factory has its own attesters but the users account will also call trust attesters if the registry provided is not address(0). so this factory behaves pretty much like the whitelist that was there before but the whitelist is in the registry using the attesters set by the owner of the factory

Reply:
initData in createAccount() allows attesters and threshold to be provided externally. how does it align with attesters and threshold (dynamic cause of ownable) stored in the contract's storage?

Reply:
So this specific factory will use the factory-native attesters to query the initial modules and then set trustAttesters with the user provided ones and from then on use this